src/topology/sheaves/sheaf_condition/unique_gluing.lean

```diff
@@ -221,22 +220,79 @@ end presheaf
 
 namespace sheaf
 open presheaf
+open category_theory
 
 variables {X : Top.{u}} (F : sheaf (Type u) X) {ι : Type u} (U : ι → opens X)
 
 /--
 A more convenient way of obtaining a unique gluing of sections for a sheaf
 -/
-lemma exists_unique_gluing (sf : Π i : ι, F.presheaf.obj (op (U i)))
- (hsf : is_compatible F.presheaf U sf) :
- ∃! s : F.presheaf.obj (op (supr U)), is_gluing F.presheaf U sf s  :=
+lemma exists_unique_gluing (sf : Π i : ι, F.presheaf.obj (op (U i))) :
+  is_compatible F.presheaf U sf →
+  ∃! s : F.presheaf.obj (op (supr U)), is_gluing F.presheaf U sf s := λ sf_comp,
 begin
-  have := (sheaf_condition_unique_gluing_of_sheaf_condition _ F.sheaf_condition U sf hsf),
+  have := (sheaf_condition_unique_gluing_of_sheaf_condition _ F.sheaf_condition U sf sf_comp),
   refine ⟨this.default.1,this.default.2,_⟩,
   intros s hs,
   exact congr_arg subtype.val (this.uniq ⟨s,hs⟩),
 end
 
+/--
+In this version of the lemma, the inclusion homs `iUV` can be specified directly by the user,
+which can be more convenient in practice.
+-/
+lemma exists_unique_gluing' (V : opens X) (iUV : Π i : ι, U i ⟶ V)
+  (hcover : V ≤ supr U) (sf : Π i : ι, F.presheaf.obj (op (U i))) :
+  is_compatible F.presheaf U sf →
+  ∃! s : F.presheaf.obj (op V), ∀ i : ι, F.presheaf.map (iUV i).op s = sf i := λ sf_comp,
+begin
+  have V_eq_supr_U : V = supr U := le_antisymm hcover (supr_le (λ i, le_of_hom (iUV i))),
+  obtain ⟨gl, gl_spec, gl_uniq⟩ := F.exists_unique_gluing U sf sf_comp,
+  refine ⟨F.presheaf.map (eq_to_hom V_eq_supr_U).op gl, (λ i,_), (λ gl' gl'_spec,_)⟩,
+  { rw ← functor_to_types.map_comp_apply,
+    exact gl_spec i },
+  { convert congr_arg _ (gl_uniq (F.presheaf.map (eq_to_hom V_eq_supr_U.symm).op gl') (λ i,_)) ;
+      rw ← functor_to_types.map_comp_apply,
+    { exact (functor_to_types.map_id_apply _ _).symm },
+    { convert gl'_spec i }}
+end
+
+lemma eq_of_locally_eq (s t : F.presheaf.obj (op (supr U))) :
+  (∀ i : ι, F.presheaf.map (opens.le_supr U i).op s = F.presheaf.map (opens.le_supr U i).op t)
+  → s = t :=
+begin
+  intro h,
```

Comment:
Why not just move `h` before the colon?

Reply:
Do you mean before the `:=` as in `:= λ h, begin...`? Or move it before the colon as in `...(h : ...) : s = t`? I was thinking since the implication is non-dependent, I'd rather make that clear and don't bother giving the hypothesis a name

Reply:
Okay I just checked the library style guidelines and it says that moving it before the colon is preferred, so I'm gonna change it